### PR TITLE
ADEN-6533 Implement N+1 DTC model

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/*global define*/
+define('ext.wikia.adEngine.ml.model.decisionTreeClassifier', [
+	'wikia.log'
+], function () {
+	'use strict';
+
+	function buildUrl(modelId) {
+		return '/wikia.php?controller=AdEngine2Api&method=getModelData&id=' + modelId
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js]: Shall we add a semicolon at the end of this line?

File: extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	}
+
+	function findMax(nums) {
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js]: How does it differ from [`Math.max()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max)?

Comment (fraszczakszymon) [reply extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js]: It returns an index

File: extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js (hunk 1 above)
Comment (fraszczakszymon) [reply extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js]: Definitely :+1:

File: extensions/wikia/AssetsManager/config.php [hunk 3]
```diff
@@ -954,14 +973,6 @@
 	]
 ];
 
-//TODO: remove after ADEN-5624 release + 24h (2 Aug 2017)
```
Comment (nandy-andy) [inline extensions/wikia/AssetsManager/config.php:957]: :gem:

File: extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js (hunk 2 above)
Comment (nandy-andy) [reply extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js]: :man_facepalming: you're right ;-)

File: extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+define('ext.wikia.adEngine.ml.model.decisionTreeClassifier', [
+	'wikia.log'
+], function () {
+	'use strict';
+
+	function buildUrl(modelId) {
+		return '/wikia.php?controller=AdEngine2Api&method=getModelData&id=' + modelId;
+	}
+
+	function findMax(nums) {
```
Comment (v3rth) [inline extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js]: What do you think about `nums.indexOf(Math.max(nums))`?

**Review by v3rth — APPROVED** [review]

File: extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+					thresholds = params.thresholds;
+					indices = params.indices;
+					classes = params.classes;
+
+					callback(params);
+				}
+			};
+
+			request.open('GET', buildUrl(modelId), true);
+			request.send();
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js]: Did you consider using `$.nirvana.sendRequest()`? I have a feeling that we're reinventing a wheel here :)

File: extensions/wikia/AdEngine/js/ml/n1/n1DecisionTreeClassifier.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+			inputParser: inputParser,
+			model: dtc,
+			name: 'n1dtc',
+			wgCountriesVariable: 'wgAdDriverN1DecisionTreeClassifierRabbitCountries',
+			enabled: false
+		};
+
+	if (geo.isProperGeo(instantGlobals[modelData.wgCountriesVariable])) {
+		dtc.loadParameters(function () {
+			modelData.enabled = true;
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/ml/n1/n1DecisionTreeClassifier.js:22]: Should we track if somewhere if the model isn't ready yet when we need it?

File: extensions/wikia/AdEngine/js/ml/n1/n1DecisionTreeClassifierInputParser.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+		log(['N+1 DTC data', data], log.levels.debug, logGroup);
+
+		return data;
+	}
+
+	return {
+		getData: getData
+	};
+});
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/ml/n1/n1DecisionTreeClassifierInputParser.js]: missing new line

File: extensions/wikia/AdEngine/js/ml/n1/n1DecisionTreeClassifier.js (hunk 6 above)
Comment (fraszczakszymon) [reply extensions/wikia/AdEngine/js/ml/n1/n1DecisionTreeClassifier.js:22]: We'll have that info (missing) in the adengadinfo table

File: extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js (hunk 4 above)
Comment (fraszczakszymon) [reply extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js]: Discussed offline

File: extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js (hunk 5 above)
Comment (fraszczakszymon) [reply extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js]: There is no $.nirvana on mobile-wiki :(

**Review by kvas-damian — APPROVED** [review]

**Pull request merged**